### PR TITLE
chore(NODE-6177): set up tooling for new repo

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- Adds a new script that manages obtaining libmongocrypt and building it
- Fixes changelog

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

- Libmongocrypt is now versioned by the "mongodb:libmongocrypt" package spec

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .github/scripts/libmongocrypt.mjs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  await run('git', ['fetch', '--tags'], { cwd: libmongocryptRoot });
+  await run('git', ['checkout', args.libmongocrypt.ref, '-b', `r-${args.libmongocrypt.ref}`], { cwd: libmongocryptRoot });
+} else {
+  console.error('libmongocrypt already up to date...', args.libmongocrypt);
+}
+
+const libmongocryptBuiltVersion = await fs.readFile(path.join(libmongocryptRoot, 'VERSION_CURRENT'), 'utf8').catch(() => '');
+const libmongocryptAlreadyBuilt = libmongocryptBuiltVersion.trim() === args.libmongocrypt.ref;
+
+if (!args.clean && !libmongocryptAlreadyBuilt) {
```
Comment (baileympearson) [inline .github/scripts/libmongocrypt.mjs:68]:
```suggestion
if (args.clean || !libmongocryptAlreadyBuilt) {
```

Am I misunderstanding clean?  I assumed that when clean is true, it force installs and builds libmongocrypt

Comment (baileympearson) [reply .github/scripts/libmongocrypt.mjs:68]: (same comment for the other if-statement as well)

**Review by baileympearson — CHANGES_REQUESTED** [review]
just a small comment

File: .github/scripts/libmongocrypt.mjs (hunk 1 above)
Comment (nbbeeken) [reply .github/scripts/libmongocrypt.mjs]: Fixed!

**Review by baileympearson — APPROVED** [review]

**Pull request merged**